iOS Client/Store/MBStore.swift

```diff
@@ -78,11 +78,65 @@ class MBStore: NSObject {
     
 
     //New function for devotions until we figure out if this will have to go over network or can be stored locally
-    func syncDevotions(completion: @escaping ([MBDevotion]?, Error?) -> Void) {
-        loadDevotions { (devotions, devotionError) in
-            completion(devotions, devotionError)
+    func syncDevotions(completion: @escaping ([LoadedDevotion]?, Error?) -> Void) {
```

Comment:
I think this is still too complex. So conceptually, the `sync` part of this seems to be syncing the devotions data stored locally with the readDates file. (There is some notion of syncing with the external store, but really that's a lazy fetch of the devotions data in case we don't have it locally)

Despite a potential performance hit, I propose we do away with the extra readDatesFile and just store the array of LoadedDevotion objects. This means that marking a devotion as read is reading out the entire array of all devotions, finding the one of interest and marking it as read, then saving the array back to disk. I think that's better than maintaining two files.

I then think this method could be decomposed into two. `getDevotions` that blindly reads LoadedDevotions from documents directory and synchronously returns them, and `syncDevotions` that pulls them from the external store, converts them to LoadedDevotions, and saves them to documents directory (as well as returns them in a completion block) as you've done here. 

Then, everywhere in our app we need devotions, we can just call `getDevotions` without needing a completion block. How the sync initially gets kicked off is up for discussion. Maybe our state will let us know if they've been loaded yet. Maybe getDevotions can just return an empty list and kick off a sync job that triggers a store action on completion. Not entirely sure, but I do think it's important to

1) eliminate the extra readDatesFile and
2) create a synchronous `getDevotions`